settings/resource_notification_destination.go

```diff
@@ -0,0 +1,117 @@
+	s.SchemaPath("config", "generic_webhook", "url_set").SetComputed()
+	s.SchemaPath("config", "generic_webhook", "password_set").SetComputed()
+	s.SchemaPath("config", "generic_webhook", "username_set").SetComputed()
+
+	// ForceNew fields
+	s.SchemaPath("destination_type").SetForceNew()
+
+	// ConflictsWith fields
+	config_eoo := []string{"config.0.slack", "config.0.pagerduty", "config.0.microsoft_teams", "config.0.generic_webhook", "config.0.email"}
+	s.SchemaPath("config", "slack").SetExactlyOneOf(config_eoo)
+	// s.SchemaPath("config", "pagerduty").SetExactlyOneOf(config_eoo)
+	// s.SchemaPath("config", "microsoft_teams").SetExactlyOneOf(config_eoo)
+	// s.SchemaPath("config", "generic_webhook").SetExactlyOneOf(config_eoo)
+	// s.SchemaPath("config", "email").SetExactlyOneOf(config_eoo)
+
+	// RequiredWith fields
+	s.SchemaPath("config", "slack").SetRequiredWith([]string{"config.0.slack.0.url"})
+	s.SchemaPath("config", "pagerduty").SetRequiredWith([]string{"config.0.pagerduty.0.integration_key"})
+	s.SchemaPath("config", "microsoft_teams").SetRequiredWith([]string{"config.0.microsoft_teams.0.url"})
+	s.SchemaPath("config", "generic_webhook").SetRequiredWith([]string{"config.0.generic_webhook.0.url"})
+	s.SchemaPath("config", "email").SetRequiredWith([]string{"config.0.email.0.addresses"})
+
+	// s.SchemaPath("config", "slack", "url").SetRequiredWith([]string{"config.0.slack"})
+	// s.SchemaPath("config", "pagerduty", "integration_key").SetRequiredWith([]string{"config.0.pagerduty"})
+	// s.SchemaPath("config", "microsoft_teams", "url").SetRequiredWith([]string{"config.0.microsoft_teams"})
+	// s.SchemaPath("config", "generic_webhook", "url").SetRequiredWith([]string{"config.0.generic_webhook"})
+	// s.SchemaPath("config", "generic_webhook", "password").SetRequiredWith([]string{"config.0.generic_webhook"})
+	// s.SchemaPath("config", "email", "addresses").SetRequiredWith([]string{"config.0.email"})
+
+	// Sensitive fields
+	s.SchemaPath("config", "generic_webhook", "password").SetSensitive()
+
+	return s
+}
+
+var ndSchema = common.StructToSchema(NDStruct{}, nil)
+
+func ResourceNotificationDestination() common.Resource {
+	return common.Resource{
+		Schema: ndSchema,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var newNDrequest settings.CreateNotificationDestinationRequest
+			common.DataToStructPointer(d, ndSchema, &newNDrequest)
+			createdND, err := w.NotificationDestinations.Create(ctx, newNDrequest)
+			if err != nil {
+				return err
+			}
+			d.SetId(createdND.Id)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			readND, err := w.NotificationDestinations.Get(ctx, settings.GetNotificationDestinationRequest{
+				Id: d.Id(),
+			})
```

Comment:
I understand your points but I think both cases are a little different.
-> in the case you mentioned, sensitive fields are supposed to be returned by the `get` call
-> In out case, Only the boolean confirmations of whether the fields are set or not are sent (no sensitive info)

On a side note, while I was looking at the mlflow_webhook resource, I could not find where did they specify the `access_tokens` as sensitive? It's not done in the SDK and neither in provider

Reply:
It's a bit more complicated because now if we think of it, even if user changes anything in configuration, it will not be detected (as no useful information is returned in `get`). So, we will need to use `md5` as we used in dashboard resource

Reply:
regarding mlfow webhook - most probably it should be fixed.

regarding the Read operation - the Get won't bring the data back, and as result, StructToData will fill the parameters with empty values, so when you run `terraform plan` after apply, it should show you the diff

Reply:
This is what I'm talking about - this is after I did `apply`, but `plan` still shows changes:

```
  # databricks_notification_destination.slack will be updated in-place
  ~ resource "databricks_notification_destination" "slack" {
        id               = "acf3f051-d075-42f6-99b0-e87768045030"
        # (2 unchanged attributes hidden)

      ~ config {
          ~ slack {
              + url     = "https://hooks.slack.com/services/..."
                # (1 unchanged attribute hidden)
            }
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

Reply:
Got it. Also, can you tell me how can I test it against any web hooks so that I can check for the things locally

Reply:
I tried doing the same thing but in step 3, I would require to set (say, `slack_url`) a nested field. For that, I tried doing
d.Set("config.0.slack.0.url", <name_of_temp_struct_in_step_1>.Config.SlackConfig.Url)
but it cannot be set like this, it throws the following error
[ERROR] setting state: config.0.slack.0.url: can only set full list
I also tried doing:
d.Set("config.0.slack", <name_of_temp_struct_in_step_1>.Config.SlackConfig), it also throws the same error.
Even I cannot set "config" directly.

Reply:
You have a structure that you already read via API - just fill missing fields in it, and that's all - you don't need to use `d.Set`